cmd/bosun/expr/expr_test.go

```diff
@@ -224,3 +224,31 @@ func TestQueryExpr(t *testing.T) {
 		}
 	}
 }
+
+func TestScalarPromotion(t *testing.T) {
+	tests := map[string]map[string]Value{
+		`abs(-1)`: {"": Number(1)},
+	}
+
+	for exprText, expected := range tests {
+		e, err := New(exprText)
+		if err != nil {
+			t.Fatal(err)
+		}
+		results, _, err := e.Execute(nil, nil, nil, nil, nil, queryTime, 0, false, nil, nil, nil)
```

Comment:
I'm not convinced that's a great test. The scalar promotion stuff does things with groups, and I want to make sure it works at the top level.